### PR TITLE
feat: add SystemSchemaProvider to QueryExecutor

### LINKED ISSUE #24986: Tie system tables queries into the `QueryDatabase` in the `query_executor` using core traits/types
Use the `IoxSystemTables` trait introduced to core in https://github.com/influxdata/influxdb3_core/pull/18 to implement system tables support for `influxdb3`.

This issue is just for getting the framework in place, actual system table implementation is detailed in other issues under the parent.

### PR DESCRIPTION
**Pull request by hiltontj**
A shell for the `system` table provider was added to the `QueryExecutorImpl` which currently does not do anything, but will enable us to tie the different system table providers into it. For example, see the follow up PR that adds the `system.queries` table: https://github.com/influxdata/influxdb/pull/24992

One notable change in this PR was the elevation of the `QueryLog` from the `Database`, i.e., namespace provider, to the `QueryExecutorImpl`, so that it lives across queries.

Closes #24986

### CODE REVIEW

File: influxdb3_server/src/query_executor.rs [hunk 1]
```diff
@@ -486,7 +495,6 @@ impl<B: WriteBuffer> QueryTable<B> {
         filters: &[Expr],
         _limit: Option<usize>,
     ) -> Result<Vec<Arc<dyn QueryChunk>>, DataFusionError> {
-        // TODO - this is only pulling from write buffer, and not parquet?
```
Comment (hiltontj) [inline influxdb3_server/src/query_executor.rs:489]: This comment seems irrelevant.

**Review by pauldix — APPROVED** [review]
